applications/asset_tracker_v2/src/modules/data_module.c

```diff
@@ -964,6 +964,12 @@ static void agps_request_handle(struct nrf_modem_gnss_agps_data_frame *incoming_
 {
 	int err;
 
+#if defined(CONFIG_NRF_CLOUD_AGPS) && defined(CONFIG_NRF_CLOUD_PGPS)
```

Comment:
If both P-GPS and A-GPS are enabled at the same time, A-GPS will be prioritized over P-GPS. Only if A-GPS fails, P-GPS will be attempted. I'm not sure if these changes make sense considering this. A-GPS will most likely succeed and then ephemeride/almanac will never be requested.

Also, the standard behavior in case P-GPS is enabled is to disable A-GPS. Done in `src/modules/Kconfig.cloud_module`.

Is there a scenario where we would want to enable both A and P -GPS at the same time?

Reply:
As far as I understand, the idea when both A-GPS and P-GPS are enabled should be that A-GPS is used to supply all assistance data except ephemerides and almanacs and P-GPS is used to supply ephemerides. Assuming that valid ephemerides are always available, almanacs are not important. But based on the function description and the implementation this doesn't seem to work like that, there's a return after the `nrf_cloud_agps_request()` call and the only `nrf_cloud_pgps_notify_prediction()` call is after that.

It looks like ATv2 wouldn't support both A-GPS and P-GPS at the same time?

Reply:
Yes. This is true. ATv2 does not currently support A and P -GPS enabled at the same time. If it makes sense to have them coexist we should definitely do this. This will mainly impose changes in the `data` and `cloud` modules to have this sorted out.

Reply:
This will probably not make it to v1.8.0, because the content freeze is tomorrow? So I'm thinking whether the zeroing of `sv_mask_ephe` and `sv_mask_alm` should be removed from this PR. It can be added later with CIA-460.